### PR TITLE
feat: warn for incompat nextauth url and port

### PR DESCRIPTION
**Pull request by juliusmarminge**
Supersedes #1591

### CODE REVIEW

**Comment by juliusmarminge** [comment]
Current dependencies on/for this PR:
* next
  * **PR #1596** <a href="https://app.graphite.dev/github/pr/t3-oss/create-t3-app/1596" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/t3-oss/create-t3-app/1596?utm_source=stack-comment).

File: cli/template/extras/config/next-config/with-auth-appdir.mjs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { env } from "./src/env.mjs";
+
+// Warn if local development URL doesn't match with the active port
+if (
+  process.env.NODE_ENV === "development" &&
+  !env.NEXTAUTH_URL.includes(String(env.PORT))
+) {
+  console.warn(
+    `NEXTAUTH_URL (${env.NEXTAUTH_URL}) doesn't match with PORT (${env.PORT})`
+  );
```
Comment (juliusmarminge) [inline cli/template/extras/config/next-config/with-auth-appdir.mjs:14]: is logging the right thing or should it be a hard throw?

Comment (c-ehrlich) [reply cli/template/extras/config/next-config/with-auth-appdir.mjs:14]: imo throw. logs are easy to miss, and if someone really wants to develop with a port that doesn't match their nextauth url (can't imagine this) they can always turn it off.

**Comment by juliusmarminge** [comment]
CleanShot 2023-10-16 at 08 32 57@2x

Changed to erroring @c-ehrlich

**Comment by juliusmarminge** [comment]
Hmm looks like `process.env.PORT` isn't always defined which causes some issues 🤔 

CleanShot 2023-10-16 at 09 26 10@2x

**Comment by c-ehrlich** [comment]
closing for now since it doesn't seem possible to achieve this without having it run on every request. would still be nice to find a way to do this in the future / if next changes their behaviour on this

**Pull request closed without merging**